### ISSUE TITLE
PHP Deprecated error running Composer-Setup.exe or .php

### ISSUE THREAD

**Issue by W1DJM** [composer/composer#9111]
Following the instructions found [here](https://getcomposer.org/download/) while using PHP 8 Beta 1 I am getting a PHP Deprecated error for the openssl_free_key() function. From the command line using:

php composer-setup.php --install-dir=d:\composer

The complete error shown is:

Deprecated: Function openssl_free_key() is deprecated in D:\composer\composer-setup.php on line 1034

From the command line the phar file DOES get created. Using Composer-Setup.exe fails with the same error but no phar file is created.

I expected the installation to complete without errors. At least with the command line method I do get a phar file but not with the GUI version.

![Composer_details](https://user-images.githubusercontent.com/11816993/89713982-78d15680-d969-11ea-9b56-d75fa151f0b7.png)
![Composer_error](https://user-images.githubusercontent.com/11816993/89713983-7969ed00-d969-11ea-8a91-45d5848ff4ad.png)

**Comment by stof** [comment]
this relates to the installer script of https://github.com/composer/windows-setup
/cc @johnstevenson

**Comment by johnstevenson** [comment]
@stof Indeed, my logic didn't cater for stderr output and a zero (success) exit code, so I'll fix this in Composer-Setup.exe.

The php installer script is the same as https://github.com/composer/getcomposer.org/blob/master/web/installer so I'll fix these to handle the deprecation in PHP 8.
